File: python/27.py
'''
Script: Project Euler problem 26
James Philbrick, July 2022

a: -999  -> 999
b: -1000 -> 1000
'''

def is_prime(n):
    # implementation of complexity O(sqrt(N))
    if n <= 1:
        return False
    i = 2
    while i**2 <= n:
        if n % i == 0:
            return False
        i += 1
    return True

def get_len_consec_primes(a, b):
    lenConsecPrimes = 0
    n = 0
    while True:
        if is_prime(n**2 + a*n + b):
            lenConsecPrimes += 1
            n += 1
        else:
            break
    return lenConsecPrimes

def main():
    maxLenConsecPrimes = 0
    coeffProduct = 0
    for a in range(-999, 1000):
        for b in range(-1000, 1001):
            lenConsecutivePrimes = get_len_consec_primes(a, b)
            if lenConsecutivePrimes > maxLenConsecPrimes:
                maxLenConsecPrimes = lenConsecutivePrimes
                coeffProduct = a*b
            else:
                pass
    print(coeffProduct)

if __name__ == '__main__':
    main()

File: go/27.go
package main

import "fmt"

func main() {
	var maxLenConsecPrimes int = 0
	var coeffProduct int = 0
	for a := -999; a < 999; a++ {
		for b := -1000; b < 1000; b++ {
			var lenConsecPrimes int = get_len_consec_primes(a, b)
			if lenConsecPrimes > maxLenConsecPrimes {
				maxLenConsecPrimes = lenConsecPrimes
				coeffProduct = a * b
			}
		}
	}
	fmt.Println(coeffProduct)
}

func is_prime(n int) bool {
	if n <= 1 {
		return false
	}
	var i int = 2
	for i*i <= n {
		if n%i == 0 {
			return false
		}
		i += 1
	}
	return true
}

func get_len_consec_primes(a int, b int) int {
	var lenConsecPrimes, n int = 0, 0
	for true {
		if is_prime(n*n + a*n + b) {
			lenConsecPrimes += 1
			n += 1
		} else {
			break
		}
	}
	return lenConsecPrimes
}
